src/Name/Generator/CommandNameGenerator.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Jaeger\Symfony\Name\Generator;
+
+use Symfony\Component\Console\Event\ConsoleCommandEvent;
+use Symfony\Component\Console\Event\ConsoleTerminateEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class CommandNameGenerator implements NameGeneratorInterface, EventSubscriberInterface
+{
+    private array $generators;
+
+    private string $name;
+
+    public function __construct(array $generators = [])
+    {
+        $this->generators = $generators;
+    }
+
+    public function add(string $regexp, NameGeneratorInterface $generator): CommandNameGenerator
+    {
+        $this->generators[sprintf('/%s/', $regexp)] = $generator;
```

Comment:
Should we use `preg_quote`?

Or - would it better, if we won't wrap string by own symbols? From the contract (interface, method signature), it is not clear, that the user should not specify real regexp, but only "regexp body"?

Reply:
Premature. I don't see the reason in over complicating feature that is not field tested yet. This is just for the convenience purposes - do not put / / on each line of configuration. For better clarity we can move this into the configuration parsing though

Reply:
Then it should be mentioned in method signature/phpdoc/docs - "this parameter is not truly regexp, method expect only 'regexp pattern'"

Reply:
addded respective lines in the docs, moved parsing to the JaegerExtension